### ISSUE TITLE
Making sub_block lazy loading optional

### ISSUE THREAD

**Issue by ginty** [Origen-SDK/origen#200]
I'm having issues with some apps where callbacks, particularly startup/on_create are no longer being called as the app originally designed them due to sub-blocks not being instantiating at definition time. This is causing an un-intended backwards compatibility issue with the latest versions of Origen.

To fix this, I would like to propose the following change and would welcome feedback on it...

- Rollback sub_block definitions to make them immediately instantiate the associated model/controller by default, i.e. restore the original behavior
- Add in a new sub_block option, `lazy: true`, which will turn on the lazy instantiation

The rationale for making the default be full instantiation is as follows:

- It is the safest, since application bugs (potentially hard to detect) can be introduced when callbacks are missed. It forces engineers to opt-in and acknowledge that given sub-blocks will not instantiate straight away. They can consider the ramifications of that for any callbacks.
- The real benefit of lazy loading is when dealing with huge numbers of sub-blocks, which by definition would not be written by hand and would be generated by importers. Such models would not have any callbacks anyway and the importer/exporter can globally apply the lazy switch.
- The Origen exporter will be updated to apply `lazy: true` by default.

In short, this means that imported/generated sub-blocks will continue to pick up the benefits of lazy instantiation by default, while user written sub-blocks will listen for callbacks by default, which seems like the best of both worlds.

Let me know if you agree/disagree...

**Comment by info-rchitect** [comment]
@ginty seems fine to me.  will try to get this done this week.

**Comment by pderouen** [comment]
@ginty, I agree.

**Comment by priyavadan** [comment]
@ginty, I agree...

**Comment by welguisz** [comment]
@ginty seems fine to me.

**Comment by info-rchitect** [comment]
https://github.com/Origen-SDK/origen/pull/203
